src/lib/marshal/translation/reader.ts

```diff
@@ -47,13 +52,30 @@ const readTranslationFiles = async (
     // push commands. Consider making this an option in the future.
     if (namespace === SYSTEM_NAMESPACE) continue;
 
-    // eslint-disable-next-line no-await-in-loop
-    const [content, readJsonErrors] = await readJson(abspath);
-
-    if (readJsonErrors.length > 0) {
-      const e = new SourceError(formatErrors(readJsonErrors), abspath);
-      errors.push(e);
-      continue;
+    // Get translation format from file extension
+    const format = getFormatFromFilename(abspath);
+
+    let content = "";
+    if (format === "json") {
```

Comment:
Nit, can we create a new helper function `readTranslationFile()` here that takes `abspath`, and reads the file accordingly based on its format, then return a uniform [content, sourceError]?

Something like:
```
const format = getFormatFromFilename(abspath);

switch (format) {
  case "json":
    //  read json
  case "po":
    // read file
  default:
    // raise, unsupported ${format}
}
```